pkgs/tools/text/gawk/default.nix

```diff
@@ -74,9 +45,66 @@ stdenv.mkDerivation rec {
 
     license = licenses.gpl3Plus;
 
-    platforms = platforms.unix;
+    platforms = platforms.unix ++ platforms.windows;
 
     maintainers = [ ];
   };
-}
+in
+
+if stdenv.hostPlatform.isWindows then
+  stdenv.mkDerivation rec {
+    inherit src name meta;
+
+    # $src/pc has pre-configured makefiles for Windows
+    configurePhase = ''
+      substituteInPlace pc/gawkmisc.pc    --replace 'int execvp(const char *file, const char *const *argv)' 'int execvp(const char *file, char *const *argv)'
+      substituteInPlace pc/Makefile       --replace CC=gcc               CC=${stdenv.cc.targetPrefix}cc
+      substituteInPlace pc/Makefile       --replace 'prefix = c:/gnu'    "prefix = $out"
+      substituteInPlace pc/Makefile.ext   --replace 'prefix = c:/gnu'    "prefix = $out"
+      sed -i -r 's,\tgcc ,\t${stdenv.cc.targetPrefix}cc ,g' pc/Makefile.ext
+
+      mv pc/Makefile.ext ./extension/Makefile
+      mv pc/{Makefile,config.h,popen.h,socket.h,dlfcn.h,in.h,getid.c,popen.c} ./
+      mv support/{dfa.h,random.h,getopt.h,regex.h,getopt_int.h,xalloc.h,regex_internal.h,localeinfo.h,regcomp.c,regexec.c,regex_internal.c,localeinfo.c,dfa.c,getopt.c,random.c,regex.c,getopt1.c} ./
```

Comment:
Are there any files we don't want copy in this directory?
I would prefer this part to be a little bit simpler otherwise it might start to bitrot fast.

Reply:
I am not sure.
So far we have no ```nix.exe``` so it is only for cross-compilation.
Native compilation may require changes.